nextcord/automod.py

```diff
@@ -0,0 +1,365 @@
+        exempt_channels: List[GuildChannel] = ...,
+        reason: str = ...,
+    ) -> AutoModerationRule:
+        ...
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: str = ...,
+        event_type: EventType = ...,
+        presets: List[KeywordPresetType] = ...,
+        notify_channel: GuildChannel = ...,
+        timeout_seconds: int = ...,
+        block_message: bool = ...,
+        enabled: bool = ...,
+        exempt_roles: List[Role] = ...,
+        exempt_channels: List[GuildChannel] = ...,
+        reason: str = ...,
+    ) -> AutoModerationRule:
+        ...
+
+    async def edit(
+        self,
+        *,
+        name: str = MISSING,
+        event_type: EventType = MISSING,
+        keyword_filters: List[str] = MISSING,
+        presets: List[KeywordPresetType] = MISSING,
+        notify_channel: GuildChannel = MISSING,
+        timeout_seconds: int = MISSING,
+        block_message: bool = MISSING,
+        enabled: bool = MISSING,
+        exempt_roles: List[Role] = MISSING,
+        exempt_channels: List[GuildChannel] = MISSING,
+        reason: str = MISSING,
+    ) -> AutoModerationRule:
+        """|coro|
+
+        Edit this auto moderation rule.
+
+        You must have the :attr:`~Permissions.manage_guild` to be able to do this.
+
+        Parameters
+        -----------
+        name: :class:`str`
+            The new name of this auto moderation rule.
+        event_type: :class:`EventType`
+            The new trigger event type of this auto moderation rule.
+        keyword_filters: List[:class:`str`]
+            The keywords that the filter should match.
+
+            .. note::
+
+                This will only work if the rule's ``trigger_type`` is :attr:`TriggerType.keyword`.
+        presets: List[:class:`KeywordPresetType`]
+            The keyword presets that the filter should match.
+
+            .. note::
+
+                This will only work if the rule's ``trigger_type`` is :attr:`TriggerType.keyword_presets`
+        notify_channel: :class:`abc.GuildChannel`
+            The channel that will receive the notification when this rule is triggered. Cannot be mixed with ``notify_channels``.
+        timeout_seconds: :class:`int`
+            The seconds to timeout the person that triggered this rule.
+        block_message: :class:`bool`
+            Whether or not this rule should block the message which triggered the rule.
+        enabled: :class:`bool`
+            Whether or not if this rule is enabled.
+        exempt_roles: :class:`Role`
+            A list of roles that should not be affected by this rule.
+        exempt_channels: :class:`abc.GuildChannel`
+            A list of channels that should not be affected by this rule.
+        reason: :class:`str`
+            The reason why this auto moderation rule was edited.
+
+        Raises
+        -------
+        InvalidArgument
+            You specified both ``keyword_filters`` and ``presets``.
```

Comment:
See later comment regarding checks and checking to ensure at-least one is passed